app/questionnaire/router.py

```diff
@@ -82,11 +82,30 @@ def can_access_location(
         return location.block_id in self._get_allowable_path(routing_path)
 
     def can_access_hub(self) -> bool:
-        return self._schema.is_flow_hub and all(
-            self._data_stores.progress_store.is_section_complete(SectionKey(section_id))
-            for section_id in self._schema.get_section_ids_required_for_hub()
-            if section_id in self.enabled_section_ids
-        )
+        if not self._schema.is_flow_hub:
+            return False
+
+        for section_id in self._schema.get_section_ids_required_for_hub():
+            if section_id in self.enabled_section_ids:
+                repeating_list_for_section = (
+                    self._schema.get_repeating_list_for_section(section_id)
+                )
+
+                items = (
+                    self._data_stores.list_store.get(repeating_list_for_section)
+                    if repeating_list_for_section
+                    else [None]
+                )
+
+                # Type ignore: items is an iterable as per the ListModel __iter__ method
+                for list_item_id in items:  # type: ignore
```

Comment:
Ignore would silently fail if we removed `__iter__` from `ListModel` but if we explicitly typed it and then removed `__iter__` mypy would correctly error, hence, it is safer to not ignore here.

```suggestion
                items: ListModel | list[None] = (
                    self._data_stores.list_store.get(repeating_list_for_section)
                    if repeating_list_for_section
                    else [None]
                )

                for list_item_id in items:
```